### PR TITLE
docs: refactored page links

### PR DESCRIPTION
**Pull request by Commandserver**
- [x] My pull request is made against the `dev` branch.
- [x] I have ensured that the changed library can be built on your target system. I did not introduce any platform-specific code.
- [x] I have ensured that all methods and functions are **fully documented** using doxygen style comments.
- [x] I tested my commits, by adding a test case to the unit tests if needed
- [x] I have ensured that I did not break any existing API calls.
- [x] My code follows the [coding style guide](https://dpp.dev/coding-standards.html) (if you are not sure, match the code style of existing files including indent style etc).
- [x] I have not built my pull request using AI, a static analysis tool or similar without any human oversight. Where I have generated this pull request using a tool, I have justified why this is needed.

Why? because doxygen uses the filenames to generate the links when we don't use `\page`, and that's ugly e.g. `md_docpages_04_building_a_bot.html` should be just `building-a-bot.html` shouldn't it?.

There're many ways to link a page e.g.: `[Shards](\ref clusters-shards-guilds)` or `\ref clusters-shards-guilds "Shards"`, which i've used for this pr.

I've tested all links to make sure they're working.

### CODE REVIEW

**Comment by Jaskowicz1** [comment]
So, does \ref mean references will work when we run doxygen locally? I remember you mentioning "ref" in one of my PRs but I had no idea of it.

**Comment by Commandserver** [comment]
> So, does \ref mean references will work when we run doxygen locally? I remember you mentioning "ref" in one of my PRs but I had no idea of it.
> 
> If so, I'm so all for this PR and I will be praising you forever 😆

Yes. they're better as we don't hard-code a filename. So when we rename a page, we don't have to change all links with it.

**Comment by Jaskowicz1** [comment]
> Yes. they're better as we don't hard-code a filename. So when we rename a page, we don't have to change all links with it.

This is absolutely amazing then, thank you a lot!!

File: docpages/INDEX.md [hunk 1]
```diff
@@ -36,10 +36,10 @@ You can find further releases in other architectures and formats or the source c
 * Efficient caching system for guilds, channels, guild members, roles, users
 * Sharding and clustering (Many shards, one process: specify the number of shards, or let the library decide)
 * Highly optimised ETF (Erlang Term Format) support for very fast websocket throughput (*no other C++ Discord library has this!*)
-* [Slash Commands/Interactions support](https://dpp.dev/slashcommands.html)
-* [Voice support](https://dpp.dev/soundboard.html) (sending **and** receiving audio)
+* \ref slashcommands "Slash Commands/Interactions support"
```
Comment (braindigitalis) [inline docpages/INDEX.md:39]: if we are doing this please make sure the names of the links match what they used to be . usually when doing \ref, the title of the link is always the title of the page but this breaks the sentences where we put a link in a sentence.

File: docpages/INDEX.md [hunk 2]
```diff
@@ -36,10 +36,10 @@ You can find further releases in other architectures and formats or the source c
...
+* \ref soundboard "Voice support" (sending **and** receiving audio)
 * The entire Discord API is available for use in the library
-* Stable [Windows support](https://dpp.dev/buildwindows.html)
+* Stable \ref buildwindows "Windows support"
 * Ready-made compiled packages for Windows, Raspberry Pi (ARM64/ARM7/ARMv6), Debian x86/x64 and RPM based distributions
```
Comment (braindigitalis) [inline docpages/INDEX.md:43]: (yeah like this)

**Review by braindigitalis — APPROVED** [review]

**Comment by Jaskowicz1** [comment]
Checked through every page to see if anything was missing and there isn't! I'm more than happy with this PR :)

**Pull request merged**